### PR TITLE
Add floating action button

### PR DESCRIPTION
**Pull request by Justus-Saringer**

### CODE REVIEW

File: demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                    })
+            }
+        }
+    }
+}
+
+@Composable
+private fun FloatingActionButton() {
+
+    androidx.compose.material.FloatingActionButton(
```
Comment (Stallinator) [inline demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt:49]: rename fun to FloatingActionButtonExample to get rid of import here

File: demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        onClick = { /* your code*/ }
+    ) {
+        Icon(
+            painter = painterResource(id = R.drawable.ic_baseline_date_range),
```
Comment (Stallinator) [inline demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt:53]: why with painter and not like this? Icon(Icons.Rounded.Menu, contentDescription = "Localized description")

**Review by Stallinator — COMMENTED** [review]
maybe we should include all the possible types and use it in a scaffold as well, i think good idea for sample, not sure for website though

File: demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            DemoTheme {
+                Scaffold(
+                    topBar = {
+                        TopAppBar(
+                            title = {
+                                Text(text = "Floating Action Button")
+                            })
```
Comment (Darius3001) [inline demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt:32]: Parenthesis maybe in "pyramid" style (dont know what its actually called)

File: demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt [hunk 4]
```diff
@@ -29,13 +36,14 @@ class ComposableFloatingActionButton : ComponentActivity() {
                         TopAppBar(
                             title = {
```
Comment (Stallinator) [inline demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt:37]: add one floating action button here and get rid of the scaffold in the Composable

File: demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt [hunk 5]
```diff
@@ -44,23 +52,90 @@ class ComposableFloatingActionButton : ComponentActivity() {
...
+                    top.linkTo(ExtendedFABWithIcon.bottom, margin = 40.dp)
+                    start.linkTo(parent.start)
+                    end.linkTo(parent.end)
+                },
+                text = { Text(text = "Extended FAB") },
+                onClick = { /*your code*/ }
+            )
+
+            FloatingActionButton(
+                modifier = Modifier.constrainAs(customFAB) {
```
Comment (Stallinator) [inline demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButton.kt:120]: custom fab missing in xml

File: code/button/FloatingActionButton.kt [hunk 6]
```diff
@@ -10,4 +10,16 @@ private fun FloatingActionButton() {
             tint = androidx.compose.ui.graphics.Color.Black
         )
```
Comment (Stallinator) [inline code/button/FloatingActionButton.kt:11]: delete the import of androidx.compose.material.FloatingActionButton is

**Comment by Stallinator** [comment]
dont merge, somehow does not work anymore even though i tested berfore pushing

**Review by Stallinator — COMMENTED** [review]
format some more lines (eg ComposableFloatingActionButton.kt) 
and rename activitys to [...]Activity :) rest looks nice

File: demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButtonActivity.kt [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+
+        ExtendedFloatingActionButton(
+            modifier = Modifier.constrainAs(ExtendedFABWithIcon) {
+                top.linkTo(RectangleFAB.bottom, margin = 40.dp)
+                start.linkTo(parent.start)
+                end.linkTo(parent.end)
+            },
+            text = { Text(text = "Extended FAB with Icon") },
+            onClick = { /*your code*/ },
+            icon = { Icon(Icons.Filled.Add, contentDescription = "Floating Action Button") }
```
Comment (Stallinator) [inline demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButtonActivity.kt:104]: format this pls

File: demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButtonActivity.kt [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+class ComposableFloatingActionButtonActivity : ComponentActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            DemoTheme {
+                Scaffold(
+                    floatingActionButton = {
+                        FloatingActionButton(onClick = {}) {
+                            Icon(Icons.Filled.Add, "Floating Action Button")
```
Comment (Stallinator) [inline demo/src/main/java/de/check24/demo/features/button/ComposableFloatingActionButtonActivity.kt:37]: format this pls

File: code/button/FloatingActionButton.kt [hunk 9]
```diff
@@ -1,25 +1,23 @@
...
-    ExtendedFloatingActionButton(
-        text = { Text(text = "Extended FAB with Icon") },
-        onClick = { /*your code*/ },
-        icon = { Icon(Icons.Filled.Add, contentDescription = "Floating Action Button") }
-    )
-}
+        ExtendedFloatingActionButton(
+            text = { Text(text = "Extended FAB with Icon") },
+            onClick = { /*your code*/ },
+            icon = { Icon(Icons.Filled.Add, contentDescription = "Floating Action Button") }
```
Comment (Stallinator) [inline code/button/FloatingActionButton.kt:21]: format this pls

**Review by Stallinator — APPROVED** [review]

**Pull request merged**